packages/runtime/container-runtime/src/gc/garbageCollection.ts

```diff
@@ -333,6 +331,10 @@ export class GarbageCollector implements IGarbageCollector {
 				return {};
 			}
 
+			// Note that the base GC details are returned even if GC is disabled. This is to handle the special scenario
```

Comment:
@markfields I am not a fan of this - Basically, I moved the complexity of running full tree summary in one particular scenario from here to summarizer node with GC.
The scenario being - GC state exists in base snapshot but GC is disabled via a feature flag.

The reason I moved the logic is to decouple summarization from GC - before this change, summarization would consult GC to determine if fullTree summary is needed.

Any thoughts on this?

Reply:
I thought we can remove this, now that gc is only disabled if it's ALWAYS been disabled, so we know there won't be any base GC details?

Reply:
Removed the option to disable GC as per our discussion so this is not needed anymore.